src/NexusMods.CrossPlatform/Process/ProcessFactory.cs

```diff
@@ -9,18 +12,93 @@ namespace NexusMods.CrossPlatform.Process;
+
+        _processLogsFolder.CreateDirectory();
+
+        var loggingSettings = settingsManager.Get<LoggingSettings>();
+        var retentionSpan = loggingSettings.ProcessLogRetentionSpan;
+
+        var filesToDelete = _processLogsFolder
+            .EnumerateFiles()
+            .Where(x => DateTime.Now - x.FileInfo.CreationTime >= retentionSpan)
+            .ToArray();
+
+        foreach (var file in filesToDelete)
+        {
+            try
+            {
+                file.Delete();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to delete expired log file {Path}", file);
+            }
+        }
     }
 
     /// <inheritdoc />
-    public async Task<CommandResult> ExecuteAsync(Command command,
+    public async Task<CommandResult> ExecuteAsync(
+        Command command,
+        bool logProcessOutput = true,
         CancellationToken cancellationToken = default)
+    {
+        if (!logProcessOutput)
+        {
+            return await ExecuteAsync(command, cancellationToken);
+        }
+
+        string fileName;
+        if (PathHelpers.IsRooted(command.TargetFilePath, _fileSystem.OS))
+        {
+            fileName = _fileSystem.FromUnsanitizedFullPath(command.TargetFilePath).FileName;
+        }
+        else
+        {
+            fileName = new RelativePath(command.TargetFilePath).FileName.ToString();
+        }
+
+        var logFileName = $"{fileName}-{DateTime.Now:s}";
+        var stdOutFilePath = _processLogsFolder.Combine(logFileName + ".stdout.log");
+        var stdErrFilePath = _processLogsFolder.Combine(logFileName + ".stderr.log");
+        _logger.LogInformation("Using process logs {StdOutLogPath} and {StdErrLogPath}", stdOutFilePath, stdErrFilePath);
+
+        await using (var stdOutStream = stdOutFilePath.Open(FileMode.Create, FileAccess.ReadWrite, FileShare.Read))
+        await using (var stdErrStream = stdErrFilePath.Open(FileMode.Create, FileAccess.ReadWrite, FileShare.Read))
+        {
+            var stdOutPipe = PipeTarget.ToStream(stdOutStream, autoFlush: true);
+            var stdErrPipe = PipeTarget.ToStream(stdErrStream, autoFlush: true);
+
+            var mergedStdOutPipe = command.StandardOutputPipe == PipeTarget.Null ? stdOutPipe : PipeTarget.Merge(command.StandardOutputPipe, stdOutPipe);
+            var mergedStdErrPipe = command.StandardErrorPipe == PipeTarget.Null ? stdErrPipe : PipeTarget.Merge(command.StandardErrorPipe, stdErrPipe);
+
+            command = command
+                .WithStandardOutputPipe(mergedStdOutPipe)
+                .WithStandardErrorPipe(mergedStdErrPipe);
+
+            var result = await ExecuteAsync(command, cancellationToken);
+            _logger.LogInformation("Command `{Command}` finished after {RunTime} seconds with exit Code {ExitCode}", command.ToString(), result.RunTime.TotalSeconds, result.ExitCode);
+
+            return result;
```

Comment:
Same here, want to refactor this for clarity? If not ok anyways

Reply:
Not much to refactor.